### PR TITLE
feature/S4-135 - Download PDF from confirmation page

### PR DESCRIPTION
**Pull request by peadarkelly**
dissolution-web - https://github.com/companieshouse/dissolution-web/pull/77
docker-chs-development - https://github.com/companieshouse/docker-chs-development/pull/34

# Description

* PDF link on confirmation page now redirects to certificate download controller, which will generate a signed S3 url and redirect the user directly to S3.
* Using pattern similar to approve certificate page where a custom session model is created for the new controller, and the redirect controller sets this session property

## Checklist

- [x] 3 Amigos
- [x] Acceptance Criteria met
- [x] Branch named {feature|hotfix|task}/{S4-*}
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] All unit tests passing
- [x] WAVE accessibility testing tool ran on new or updated pages
- [x] Pulled latest master into feature branch
- [ ] Code reviewed
- [x] New Docker environment variables are consistent with those on the Rebel1 environment
- [x] If your pull request depends on any other, please link them in the description

### CODE REVIEW

File: src/models/form/dissolutionApproval.model.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  companyNumber: string,
-  applicant: string,
+  companyName: string
+  companyNumber: string
+  applicant: string
   officerType: OfficerType
-  onBehalfName?: string,
+  onBehalfName?: string
   date: string
 }
```
Comment (sapphireduffy) [inline src/models/form/dissolutionApproval.model.ts:10]: Should we add an extra line at the bottom of this file?

File: test/mappers/session/dissolutionSession.mapper.test.ts [hunk 2]
```diff
@@ -54,4 +55,23 @@ describe('DissolutionApprovalMapper', () => {
...
+      dissolution.certificate_bucket = BUCKET
+      dissolution.certificate_key = KEY
+
+      const result: DissolutionConfirmation = mapper.mapToDissolutionConfirmation(dissolution)
+
+      assert.equal(result.certificateKey, KEY)
+      assert.equal(result.certificateBucket, BUCKET)
+    })
+  })
 })
```
Comment (sapphireduffy) [inline test/mappers/session/dissolutionSession.mapper.test.ts:77]: New line here also?

File: test/services/dissolution/dissolution.service.test.ts [hunk 3]
```diff
@@ -81,14 +90,31 @@ describe('DissolutionService', () => {
...
-  it('should call dissolution api client to approve the dissolution for the provided email', async () => {
-    const email: string = 'test@email.com'
-    const body: DissolutionPatchRequest = generateDissolutionPatchRequest()
+  describe('approveDissolution', () => {
+    it('should call dissolution api client to approve the dissolution for the provided email', async () => {
+      const email: string = 'test@email.com'
+      const body: DissolutionPatchRequest = generateDissolutionPatchRequest()
 
-    when(mapper.mapToDissolutionPatchRequest(email)).thenReturn(body)
+      when(mapper.mapToDissolutionPatchRequest(email)).thenReturn(body)
```
Comment (sapphireduffy) [inline test/services/dissolution/dissolution.service.test.ts:98]: Do we need these extra spaces?

Comment (peadarkelly) [reply test/services/dissolution/dissolution.service.test.ts:98]: The tests were missing the describe block, so by introducing it everything has been indented one extra tab

File: src/models/form/dissolutionApproval.model.ts (hunk 1 above)
Comment (peadarkelly) [reply src/models/form/dissolutionApproval.model.ts]: Done - also added rule to tslint so had to fix all violations in the app

File: test/mappers/session/dissolutionSession.mapper.test.ts (hunk 2 above)
Comment (peadarkelly) [reply test/mappers/session/dissolutionSession.mapper.test.ts]: Done - also added rule to tslint so had to fix all violations in the app

File: test/services/dissolution/dissolution.service.test.ts (hunk 3 above)
Comment (sapphireduffy) [reply test/services/dissolution/dissolution.service.test.ts:98]: Ahhh ok makes sense

File: src/models/form/dissolutionApproval.model.ts (hunk 1 above)
Comment (sapphireduffy) [reply src/models/form/dissolutionApproval.model.ts]: That's good! :D

**Pull request merged**